r/R/dataset.R

```diff
@@ -141,6 +141,42 @@ Dataset$create <- function(children, schema) {
 #' @export
 names.Dataset <- function(x) names(x$schema)
 
+#' @export
+dim.Dataset <- function(x) {
+
+  if (!inherits(x$format, "ParquetFileFormat")) {
```

Comment:
Referencing `$format` implicitly checks that `x` is a FileSystemDataset, but I'd prefer to explicitly check:
```suggestion
  if (!inherits(x$format, "FileSystemDataset")) {
    stop(
      "dim() is only currently implemented for Datasets that use parquet files.",
      "Call collect() first to pull data into R.",
      call. = FALSE
    )
  }
  
  if (!inherits(x$format, "ParquetFileFormat")) {
```

Reply:
I don't think we need a separate `if` for this though. If you want to be explicit that it's a FileSystemDataset, I'd do this instead

```suggestion
  if (!(inherits(x, "FileSystemDataset") && inherits(x$format, "ParquetFileFormat"))) {
```